### PR TITLE
feat: question-date

### PR DESCRIPTION
**Pull request by pandeymangg**
## What does this PR do?

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by mattinannt** [comment]
Screenshot 2023-11-21 at 18 35 25

Can we make the UI go full width? :-)

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/DateQuestionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+interface IDateQuestionFormProps {
+  localSurvey: TSurvey;
+  question: TSurveyDateQuestion;
+  questionIdx: number;
+  updateQuestion: (questionIdx: number, updatedAttributes: any) => void;
+  lastQuestion: boolean;
+  isInValid: boolean;
+}
+
+const dateOptions = [
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/DateQuestionForm.tsx:19]: Screenshot 2023-11-21 at 18 36 50

When I already have a Date selected in the Preview, changing the formatting options on the left isn't changing the UI preview on the right. Is this possible to change?

File: packages/question-date/tailwind.config.js
Comment (mattinannt) [inline packages/question-date/tailwind.config.js:1]: I think you are now including also the tailwind preflight and as the styles are not prefixed they could interfer with the styles of the page where formbricks-surveys are embedded.
I'm not 100% sure about this but it's worth rechecking.

File: packages/question-date/vite.config.ts
Comment (mattinannt) [inline packages/question-date/vite.config.ts:1]: I would prefer if we use the same vite functionality in all packages. Would it also be possible to use lib mode configuration (only `iife` or `umd` format for filetype) in this package?

File: packages/question-date/src/vite-env.d.ts
Comment (mattinannt) [inline packages/question-date/src/vite-env.d.ts:1]: Do we need this file?

File: packages/question-date/src/main.tsx
Comment (mattinannt) [inline packages/question-date/src/main.tsx:1]: can we rename this to `index.tsx`?

File: packages/question-date/src/components/app.tsx
Comment (mattinannt) [inline packages/question-date/src/components/app.tsx:1]: I think `App` is not the best word for this as this is not a standalone webapp. Maybe name it `Date` or `Question`?

File: packages/question-date/src/assets/preact.svg
Comment (mattinannt) [inline packages/question-date/src/assets/preact.svg:1]: I think this can be removed

File: packages/question-date/public/vite.svg
Comment (mattinannt) [inline packages/question-date/public/vite.svg:1]: I think this can be removed

File: packages/question-date/index.html
Comment (mattinannt) [inline packages/question-date/index.html:1]: Can this be used for local testing or is this not needed?

**Review by mattinannt — COMMENTED** [review]
@pandeymangg thanks a lot for the feature. Looks great!!  🚀🚀🚀
Since this is now the template for sideloading questions, I think we need to iterate on this a few times to make sure we don't copy and paste things to different questions that shouldn't be copied 😉 .

**Comment by jobenjada** [comment]
hey @pandeymangg - thanks for your work today, great progress :) I've noticed a few things:

1. The widget height is not sufficient to display the calendar:

image

Any suggestions on how to solve this best without being too hacky? 🤔 

------

2. Odd "Submit" button position, label not overwritable

image

When the calendar question is the first in the survey, there are a few issues:

1. Label says Submit instead of "Next"
2. The label cannot be overwritten
3. The button is positioned on the left side instead of the right side

----

Thanks! :)

**Comment by jobenjada** [comment]
@mattinannt looks good, LGTM :)

**Review by mattinannt — APPROVED** [review]
@pandeymangg thank you for making the changes. Looks great!! 🚀
Merging it now! 😊

**Pull request merged**